### PR TITLE
Test using phantomjs

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by coolo** [comment]
selenium doesn't like devel::cover

**Comment by ancorgs** [comment]
AWESOME!

**Comment by coolo** [comment]
note that the coverage actually dropps below 70% with this as I exclude t/* now - which is stupid IMO to count as covered code

**Comment by ancorgs** [comment]
I have always been sure that our coverage stats are a plain lie, although I have never take the time to verify the reason.

**Comment by coolo** [comment]
<pre>
.../_common/deploy/_any/21-seeds.pl  100.0    n/a    n/a  100.0    0.1  100.0
lib/DBIx/Class/Timestamps.pm         100.0    n/a    n/a  100.0    5.8  100.0
lib/OpenQA.pm                         91.0   50.0   45.4   85.0    0.8   86.0
lib/OpenQA/Auth/Fake.pm              100.0   50.0    n/a  100.0    1.5   95.8
lib/OpenQA/Auth/OpenID.pm             13.4    0.0    0.0   25.0    1.5   10.7
lib/OpenQA/Auth/iChain.pm             18.7    0.0    n/a   25.0    1.4   18.1
lib/OpenQA/Client.pm                 100.0   81.2   50.0  100.0    1.0   90.3
lib/OpenQA/Controller/API/V1.pm       97.0   77.7   66.6  100.0    2.5   88.7
...penQA/Controller/API/V1/Asset.pm   97.9   83.3    n/a  100.0    2.9   96.8
...nQA/Controller/API/V1/Command.pm   69.2    n/a    n/a   75.0    2.8   70.5
lib/OpenQA/Controller/API/V1/Iso.pm   90.8   52.6   36.3   90.9    2.7   77.7
lib/OpenQA/Controller/API/V1/Job.pm   42.1   42.8    0.0   47.3    2.7   42.3
...Controller/API/V1/JobTemplate.pm   74.4   61.3   50.0   66.6    2.6   66.8
...penQA/Controller/API/V1/Table.pm   89.6   70.0  100.0  100.0    2.5   86.5
...enQA/Controller/API/V1/Worker.pm   97.0   50.0    n/a  100.0    2.5   95.4
...OpenQA/Controller/Admin/Asset.pm   42.8    n/a    n/a   50.0    2.4   44.4
.../Controller/Admin/JobTemplate.pm   15.3    0.0    0.0   50.0    2.3   14.5
...enQA/Controller/Admin/Machine.pm  100.0    n/a    n/a  100.0    2.3  100.0
...enQA/Controller/Admin/Product.pm   25.0    n/a    n/a   50.0    2.2   28.5
...QA/Controller/Admin/TestSuite.pm   25.0    n/a    n/a   50.0    2.2   28.5
lib/OpenQA/Controller/Admin/User.pm   95.4   80.0    n/a  100.0    2.2   91.4
...enQA/Controller/Admin/Workers.pm   40.9    n/a    n/a   60.0    2.1   44.4
lib/OpenQA/Controller/ApiKey.pm      100.0  100.0  100.0  100.0    1.4  100.0
lib/OpenQA/Controller/File.pm         77.4   52.7   37.5   87.5    1.3   70.5
lib/OpenQA/Controller/Running.pm      16.5    0.0    0.0   36.8    1.2   14.0
lib/OpenQA/Controller/Session.pm      62.8   46.6   66.6   76.9    1.2   60.6
lib/OpenQA/Controller/Step.pm         58.0   45.7   26.5   82.6    1.2   53.4
lib/OpenQA/Controller/Test.pm         86.8   75.0   44.8   90.0    1.1   79.9
lib/OpenQA/Controller/Worker.pm       27.2    0.0    n/a   57.1    0.9   27.1
lib/OpenQA/Plugin/CSRF.pm            100.0   66.6   33.3  100.0    0.9   91.6
lib/OpenQA/Plugin/HashedParams.pm     84.2   41.6    n/a  100.0    0.8   75.4
lib/OpenQA/Plugin/Helpers.pm          94.1   72.7   66.6  100.0    1.0   85.0
lib/OpenQA/Plugin/REST.pm             97.2   75.0   33.3  100.0    1.2   91.2
lib/OpenQA/Scheduler.pm               84.7   70.1   58.3   87.5    1.4   80.3
lib/OpenQA/Schema/Schema.pm           90.0    n/a    n/a   85.7    0.1   88.8
lib/OpenQA/Utils.pm                   57.1   40.4   80.0   71.4    4.9   55.7
lib/OpenQA/Variables.pm               52.8   25.0   14.2   88.8    0.2   48.2
lib/OpenQA/WebSockets.pm              72.8   50.0   50.0   76.9    0.1   67.3
lib/db_helpers.pm                    100.0   50.0   80.0  100.0   30.2   88.6
lib/db_profiler.pm                    29.0    n/a    n/a   37.5    0.0   30.7
Total                                 72.1   55.4   42.6   78.8  100.0   67.5
</pre>

**Comment by coolo** [comment]
rebased on #180

File: lib/OpenQA.pm [hunk 1]
```diff
@@ -39,7 +39,7 @@ sub _read_config {
             hsts => 365,
         },
         auth => {
-            method => 'OpenID',
+            method => 'Fake',
```
Comment (lnussel) [inline lib/OpenQA.pm]: Just to document offline discussion: We need to change this at least in the package. The code (not just a config file template) must default to openid there.

**Comment by coolo** [comment]
now that's depressing: a weekend of work and 59.23 -> 60.19 :)

**Comment by coolo** [comment]
I'll prepare the packaging

**Comment by coolo** [comment]
@lnussel 
https://build.opensuse.org/package/view_file/devel:openQA/openQA/set-default-auth.diff?expand=0

**Comment by aaannz** [comment]
Note that currently the auth_method name must be exactly the same as the pm file name (sans pm suffix). So in case of iChain it must be with lower i. Sorry, I didn't notice that before.

**Pull request merged**